File: library frontend/src/models/addBookRequest.ts
class AddBookRequest {
  bookTitle: string;
  bookAuthor: string;
  bookDescription: string;
  copies: number;
  bookCategory: string;
  bookImage?: string;

  constructor(
    bookTitle: string,
    bookAuthor: string,
    bookDescription: string,
    copies: number,
    bookCategory: string
  ) {
    this.bookTitle = bookTitle;
    this.bookAuthor = bookAuthor;
    this.bookDescription = bookDescription;
    this.copies = copies;
    this.bookCategory = bookCategory;
  }
}

export default AddBookRequest;

File: Library Backend/src/main/java/com/fullstack/library/management/backend/dto/AddBookRequest.java
package com.fullstack.library.management.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@NoArgsConstructor
@AllArgsConstructor
public class AddBookRequest
{
    private String bookTitle;
    private String bookAuthor;
    private String bookDescription;
    private int copies;
    private String bookCategory;
    private String bookImage;
}
